src/Surfnet/StepupGateway/GatewayBundle/Controller/SecondFactorController.php

```diff
@@ -498,10 +496,14 @@ public function verifyYubiKeySecondFactor(Request $request): Response
     public function verifySmsSecondFactor(
         Request $request,
     ): Response|RedirectResponse {
-        if (!$request->get('authenticationMode', false)) {
+        $authenticationMode = $request->query->get('authenticationMode');
+        if ($authenticationMode === '') {
+            $authenticationMode = $request->request->get('authenticationMode');
+        }
```

Comment:
Fallback to post params should not be required here, also goes for the other situations.